packages/skill-lesson/hooks/use-mux-player.tsx

```diff
@@ -106,20 +113,23 @@ export const VideoProvider: React.FC<
 
   const handleNext = React.useCallback(
     (autoPlay: boolean) => {
-      nextExerciseSlug && autoPlay
-        ? router.push({
-            pathname: '/[module]/[lesson]',
-            query: {module: moduleSlug, lesson: nextExerciseSlug},
-          })
-        : setDisplayOverlay(true)
+      if (nextExercise?._type === 'exercise') {
+        setDisplayOverlay(true)
+      } else {
+        router.push(router.asPath + '/exercise').then(() => {
+          setDisplayOverlay(true)
+        })
+      }
     },
     [moduleSlug, nextExerciseSlug, router],
   )
 
   // initialize player state
   React.useEffect(() => {
-    setDisplayOverlay(false)
-  }, [lesson])
+    router.asPath.endsWith('/exercise')
+      ? setDisplayOverlay(true)
+      : setDisplayOverlay(false)
+  }, [lesson, router.asPath])
```

Comment:
this piece of logic is responsible for displaying overlay with repl if route ends with `/exercise`